docs/source/queries/timeouts.md

```diff
@@ -0,0 +1,48 @@
+# Query timeouts
+
+Query execution time can be limited by setting a request timeout. If a query does not complete
+in the given time, then `QueryError::RequestTimeout` is returned by the driver immediately,
+so that application logic can continue operating, but the query may still be in progress on the server.
+
+As a side note, if one wishes custom server-side timeouts (i.e. actual interruption of query processing),
+one can use a[`USING TIMEOUT <duration>` directive supported in ScyllaDB](https://github.com/scylladb/scylladb/blob/master/docs/cql/cql-extensions.md#using-timeout)
+(but not in Cassandra).
+
+Timeout can be set globally (per session) or locally (for given statement).
+The default per-session timeout is currently 30s.
+It is possible to turn off timeouts completely by providing `None` as timeout when building `Session`.
+However, setting per-statement timeout to `None` results in falling back to per-session timeout.
+
+```rust
+# extern crate scylla;
+# use std::error::Error;
+# async fn timeouts() -> Result<(), Box<dyn Error>> {
+use scylla::{Session, SessionBuilder, query::Query};
+use std::time::Duration;
+
+let uri = std::env::var("SCYLLA_URI")
+    .unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+
+let session: Session = SessionBuilder::new()
+    .known_node(uri)
+    .request_timeout(None) // no per-session timeout
+    .build()
+    .await?;
+
+// We do not set the below query as idempotent, so no speculative execution gets ever fired.
+// This means that having no timeout, the query can block indefinitely if a queried node hangs.
```

Comment:
Wait what does idempotence have to do with timeouts?
Does speculative execution disable timeouts and they reset with every new attempt?

Reply:
Speculative execution makes it possible to receive a response from another node if one node hangs on CQL layer (but still sustains TCP connection). If no speculative executions and no timeouts are enabled, then in such scenario the driver would wait forever.

Reply:
With speculative execution enabled the query could also block indefinitely.

Could we remove the mention of idempotence, it really overcomplicates a simple concept.
Something like:
`With no timeout set this query could block indefinitely`
Expresses the idea without going too deep into details.

Reply:
to avoid potential bikeshedding: I'm also in favor of simplifying the docs, without mentioning idempotence